haskell/private/haskell_impl.bzl

```diff
@@ -511,6 +511,7 @@ def haskell_library_impl(ctx):
         compile_flags = c.compile_flags,
         user_compile_flags = user_compile_flags,
         user_repl_flags = _expand_make_variables("repl_ghci_args", ctx, ctx.attr.repl_ghci_args),
+        per_module_transitive_interfaces = module_outputs.per_module_transitive_interfaces,
```

Comment:
Wouldn't it be enough for this dictionary to only capture the modules that are in the `haskell_library`'s `modules` attribute? A `haskell_module` can only depend on other `haskell_module`s that are either part of the enclosing `haskell_library|...`'s `modules` attribute or contained in a direct `narrowed_deps` dependency, correct?

IIUC this could currently capture more modules than just that, namely all transitive module dependencies of all the modules in the `modules` attribute. I worry that in a project where every `haskell_library` uses `narrowed_deps` this could grow very large.

Reply:
Ah, I think I missed the "excluding modules that come from libraries in narrowed_deps" in https://github.com/tweag/rules_haskell/blob/321129c1e669adbb3b342093948c1690aa53812d/haskell/experimental/private/module.bzl#L339-L341
Makes sense then. Thank you for clarifying.